lib/block_view.go

```diff
@@ -6575,6 +6614,164 @@ func (bav *UtxoView) GetPostsPaginatedForPublicKeyOrderedByTimestamp(publicKey [
+		copy(diamondPostHash[:], key[1:1+HashSizeBytes])
+		diamondPostEntry := bav.GetPostEntryForPostHash(diamondPostHash)
+		receiverPKIDEntry := bav.GetPKIDForPublicKey(diamondPostEntry.PosterPublicKey)
+
+		senderPKID := &PKID{}
+		copy(senderPKID[:], key[1+HashSizeBytes:])
+
+		diamondKey := &DiamondKey{
+			SenderPKID:      *senderPKID,
+			ReceiverPKID:    *receiverPKIDEntry.PKID,
+			DiamondPostHash: *postHash,
+		}
+
+		loadedEntry := bav.GetDiamondEntryForDiamondKey(diamondKey)
+		_ = loadedEntry
+	}
+
+	// Iterate over the view and create the final map to return.
+	pkidToDiamondLevel := make(map[PKID]int64)
+	for _, diamondEntry := range bav.DiamondKeyToDiamondEntry {
+		if !diamondEntry.isDeleted && reflect.DeepEqual(diamondEntry.DiamondPostHash[:], postHash[:]) {
+			pkidToDiamondLevel[*diamondEntry.SenderPKID] = diamondEntry.DiamondLevel
+		}
+	}
+
+	return pkidToDiamondLevel, nil
+}
+
+func (bav *UtxoView) GetLikesForPostHash(postHash *BlockHash) (_likerPubKeys [][]byte, _err error) {
+	handle := bav.Handle
+	dbPrefix := append([]byte{}, _PrefixLikedPostHashToLikerPubKey...)
+	dbPrefix = append(dbPrefix, postHash[:]...)
+	keysFound, _ := EnumerateKeysForPrefix(handle, dbPrefix)
+
+	// Iterate over all the db keys & values and load them into the view.
+	expectedKeyLength := 1 + HashSizeBytes + btcec.PubKeyBytesLenCompressed
+	for _, key := range keysFound {
+		// Sanity check that this is a reasonable key.
+		if len(key) != expectedKeyLength {
+			return nil, fmt.Errorf("UtxoView.GetLikesForPostHash: Invalid key length found: %d", len(key))
+		}
+
+		likedPostHash := &BlockHash{}
+		copy(likedPostHash[:], key[1:1+HashSizeBytes])
+
+		likerPubKey := key[1+HashSizeBytes:]
+
+		likeKey := &LikeKey{
+			LikerPubKey:   MakePkMapKey(likerPubKey),
+			LikedPostHash: *likedPostHash,
+		}
+
+		bav._getLikeEntryForLikeKey(likeKey)
+	}
+
+	// Iterate over the view and create the final list to return.
+	likerPubKeys := [][]byte{}
+	for _, likeEntry := range bav.LikeKeyToLikeEntry {
+		if !likeEntry.isDeleted && reflect.DeepEqual(likeEntry.LikedPostHash[:], postHash[:]) {
+			likerPubKeys = append(likerPubKeys, likeEntry.LikerPubKey)
+		}
+	}
+
+	return likerPubKeys, nil
+}
+
+func (bav *UtxoView) GetRecloutsForPostHash(postHash *BlockHash) (_reclouterPubKeys [][]byte, _err error) {
+	handle := bav.Handle
+	dbPrefix := append([]byte{}, _PrefixRecloutedPostHashReclouterPubKey...)
+	dbPrefix = append(dbPrefix, postHash[:]...)
+	keysFound, _ := EnumerateKeysForPrefix(handle, dbPrefix)
+
+	// Iterate over all the db keys & values and load them into the view.
+	expectedKeyLength := 1 + HashSizeBytes + btcec.PubKeyBytesLenCompressed
+	for _, key := range keysFound {
+		// Sanity check that this is a reasonable key.
+		if len(key) != expectedKeyLength {
+			return nil, fmt.Errorf("UtxoView.GetRecloutersForPostHash: Invalid key length found: %d", len(key))
+		}
+
+		recloutedPostHash := &BlockHash{}
+		copy(recloutedPostHash[:], key[1:1+HashSizeBytes])
```

Comment:
we know the post hash before the for loop is created -- don't need to cut it out from the key.

Reply:
resolved with #16